R/init.R

```diff
@@ -183,19 +183,17 @@ init <- function(data,
   }
 
   if (inherits(data, "teal_data")) {
-    if (length(teal.data::datanames(data)) == 0) {
-      stop("`data` object has no datanames. Specify `datanames(data)` and try again.")
+    if (length(teal.data::datanames(data)) == 0 && length(names(data@env)) == 0) {
+      stop("`data` object has no datanames and its environment is empty. Specify `datanames(data)` and try again.")
     }
-
     # in case of teal_data_module this check is postponed to the srv_teal_with_splash
-    is_modules_ok <- check_modules_datanames(modules, teal.data::datanames(data))
+    is_modules_ok <- check_modules_datanames(modules, teal_data_datanames(data))
     if (!isTRUE(is_modules_ok)) {
       logger::log_error(is_modules_ok)
-      checkmate::assert(is_modules_ok, .var.name = "modules")
+      validate(need(isTRUE(is_modules_ok), is_modules_ok))
```

Comment:
init is not a shiny module, you need to hardstop here
```suggestion
      validate(need(isTRUE(is_modules_ok), is_modules_ok))
```

Reply:
you mean `checkmate::assert(is_modules_ok, .var.name = "modules")`

Reply:
Yes